doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,879 @@
+
+    2) openconfig-routing-policy.yang
+
+BGP and "routing policy" related augmented and not-supported fields are available in openconfig-bgp-ext.yang and openconfig-routing-policy-ext.yang files respectively.
+Supported YANG containers:
+```
+module: openconfig-network-instance
+    +--rw network-instances
+       +--rw network-instance* [name]
+          +--rw table-connections
+          |    ...
+          +--rw protocols
+             +--rw protocol* [identifier name]     
+                +--rw bgp
+                   +--rw global
+                   |  |   ...
+                   |  +--rw afi-safis
+                   |     +--rw afi-safi* [afi-safi-name]
+                   |        ...
+                   +--rw neighbors
+                   |  +--rw neighbor* [neighbor-address]
+                   |     |   ...
+                   |     +--rw afi-safis
+                   |        +--rw afi-safi* [afi-safi-name]
+                   |             ...   
+                   +--rw peer-groups
+                   |  +---rw peer-group* [peer-group-name]
+                   |      |  ...
+                   |      +--rw afi-safis
+                   |         +--rw afi-safi* [afi-safi-name]
+                   |                ...            
+                   +--ro rib
+                      +--ro afi-safis
+                             ...
+
+module: openconfig-routing-policy
+    +--rw routing-policy
+       +--rw defined-sets
+       |     ...
+       +--rw policy-definitions
+          +--rw policy-definition* [name]
+                ...                                             
+
+```
+### 3.6.2 CLI
+  1. For all configuration commands, the CLI request is converted to a corresponding REST client SDK request based on the Open Config data model that was generated by the Swagger generator, and is given to the REST server.
+
+  2. From there on it will follow the same path as a REST config request for create, update and delete operations.
+
+  3. The Swagger generated REST server handles all the REST requests from the client SDK and invokes a common handler for all the create, update, replace, delete and get operations along with path and payload. This common handler converts all the requests into Transformer arguments and invokes the corresponding Transformer APIs.
+
+  4. For show commands, the CLI request is converted to a corresponding REST client SDK get request based on Open Config data model's config or state object on a case by case basis.
+
+  5. For show commands that requires retrieval of the data that doesn't contain any state information (information only based on the configuration), the backend callback will fetch the data from CONFIG_DB.
+
+  6. For show commands that requires retrieval of state or statistics information the backend callback will fetch the data from FRR-BGP.
+
+  7. State/stats information is retried from FRR-BGP by issuing a show command to FRR BGP container and the output is returned in JSON format.
```

Comment:
Not in sync with flow diagram 4.2.2 - it shows translib sends a synchronous message to bgpcfgd to fetch state data.
Also, isn't direct container-to-container communication against SONiC design guidelines?

Reply:
will fix it

Reply:
This is fixed now.